### PR TITLE
tests: update tests to use LRO

### PR DESCRIPTION
**Pull request by georgiyekkert**
Update tests to use LRO.
Also manually checked LRO cancel()

### CODE REVIEW

File: google-cloud-compute/src/test/java/com/google/cloud/compute/v1/integration/ITSmokeInstancesTest.java [hunk 1]
```diff
@@ -213,20 +201,19 @@ public void testDefaultClient() throws IOException {
...
-  public void testDefaultResource() {
+  public void testDefaultResource() throws ExecutionException, InterruptedException {
     Instance instanceResource = Instance.newBuilder().build();
     try {
-      instancesClient.insert(DEFAULT_PROJECT, DEFAULT_ZONE, instanceResource);
+      instancesClient.insertAsync(DEFAULT_PROJECT, DEFAULT_ZONE, instanceResource).get();
       fail("Did not catch the exception");
-    } catch (InvalidArgumentException ex) {
-      String message = "Bad Request";
+    } catch (ExecutionException ex) {
```
Comment (vam-google) [inline google-cloud-compute/src/test/java/com/google/cloud/compute/v1/integration/ITSmokeInstancesTest.java:215]: ExecutionException is caught here, but is also declared in throws block above. Do you really need both?

**Review by vam-google — APPROVED** [review]
LGTM with few minor comments. 
As for failing tests, I think is ok - this PR merges into a branch, not into a master. And the branch it tires to merge is broken already (this is kind of the point of this PR - to fix as much as possible that broken branch)

File: google-cloud-compute/src/test/java/com/google/cloud/compute/v1/integration/ITAddressesTest.java [hunk 2]
```diff
@@ -58,7 +56,7 @@ public void setUpMethod() {
   @AfterClass
   public static void tearDown() {
     for (Address address : addresses) {
-      addressesClient.delete(DEFAULT_PROJECT, DEFAULT_REGION, address.getName());
+      addressesClient.deleteAsync(DEFAULT_PROJECT, DEFAULT_REGION, address.getName());
```
[the remark this answers is not in the document]
Comment (georgiyekkert) [reply google-cloud-compute/src/test/java/com/google/cloud/compute/v1/integration/ITAddressesTest.java]: Done.

File: google-cloud-compute/src/test/java/com/google/cloud/compute/v1/integration/ITHeadersTest.java [hunk 3]
```diff
@@ -65,7 +73,16 @@ public static void tearDown() {
 
   @Test
   public void testHeaders() {
-    addressesClient.insert("test", "test", Address.newBuilder().setName("test").build());
+    OperationFuture<Operation, Operation> future =
+        addressesClient.insertAsync("test", "test", Address.newBuilder().setName("test").build());
```
[the remark this answers is not in the document]
Comment (georgiyekkert) [reply google-cloud-compute/src/test/java/com/google/cloud/compute/v1/integration/ITHeadersTest.java]: Done

File: google-cloud-compute/src/test/java/com/google/cloud/compute/v1/integration/ITSmokeInstancesTest.java (hunk 1 above)
Comment (georgiyekkert) [reply google-cloud-compute/src/test/java/com/google/cloud/compute/v1/integration/ITSmokeInstancesTest.java:215]: We don't, removed from declaration

File: google-cloud-compute/src/test/java/com/google/cloud/compute/v1/integration/ITHeadersTest.java [hunk 4]
```diff
@@ -65,7 +73,16 @@ public static void tearDown() {
...
+    try {
+      future.get(5, TimeUnit.SECONDS);
+    } catch (InterruptedException | ExecutionException e) {
+      e.printStackTrace();
+    } catch (TimeoutException e) {; // expected
+    }
+    future.cancel(true);
```
[the remark this answers is not in the document]
Comment (georgiyekkert) [reply google-cloud-compute/src/test/java/com/google/cloud/compute/v1/integration/ITHeadersTest.java:84]: Unfortunately it doesn't, I tested it manually. e2e test for this seems a little bit more complicated, had no time to work on that.

**Pull request merged**